### PR TITLE
 drivers: spi: Add shim for nrfx SPIM driver

### PR DESCRIPTION
**Pull request by anangl**
This adds a translation layer to make the nrfx driver for the nRF SPIM
(SPI Master with EasyDMA) peripheral accessible via the Zephyr's API.
The shim is provided only for nRF52840 because of a hardware anomaly
present in nRF52832. See Anomaly 58 (SPIM: An additional byte is clocked
out when RXD.MAXCNT = 1) in Errata for this chip.

Small corrections are done to the shim for nrfx_spi driver as well,
so that similar code in these two shims is aligned and easier to
compare.

### CODE REVIEW

**Review by tbursztyka — CHANGES_REQUESTED** [review]

**Comment by anangl** [comment]
@tbursztyka I addressed your comments.

**Review by tbursztyka — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: drivers/spi/spi_nrfx_spi.c [hunk 1]
```diff
@@ -270,24 +270,12 @@ static int init_spi(struct device *dev, const nrfx_spi_config_t *config)
...
- * this peripheral. See the chapter "Peripheral interface" (sections "Peripheral
- * ID" and "Interrupts") in the product specification of a given SoC.
- */
-#define NRF_SPI_IRQ_NUMBER(idx)  (uint8_t)((uint32_t)NRF_SPI##idx >> 12u)
-
 #define SPI_NRFX_SPI_DEVICE(idx)					\
 	static int spi_##idx##_init(struct device *dev)			\
 	{								\
-		IRQ_CONNECT(NRF_SPI_IRQ_NUMBER(idx),			\
+		IRQ_CONNECT(NRFX_IRQ_NUMBER_GET(NRF_SPI##idx),		\
```
Comment (galak) [inline drivers/spi/spi_nrfx_spi.c]: we should get irq number from device tree.

Comment (anangl) [reply drivers/spi/spi_nrfx_spi.c]: There is no DTS support for nRF SPI peripherals yet. I'll add it in a next step. Or do you want me to do it in this PR?

Comment (nashif) [reply drivers/spi/spi_nrfx_spi.c]: we can do this in a followup PR.

**Pull request merged**